File: snail-race-bet/typescript/src/adapters/RaceResultProviderHttp.ts
import {RaceResultProvider, SnailRaces} from "../domain/RaceResultProvider";
import {RaceResultProviderHttpAntiCorruptionLayer} from "./RaceResultProviderHttpAntiCorruptionLayer";


export class RaceResultProviderHttp implements RaceResultProvider {
    async races(): Promise<SnailRaces> {
        const response = await fetch('http://localhost:8000/results');
        expect(response.status).toEqual(200);
        let data = await response.json();

        return RaceResultProviderHttpAntiCorruptionLayer.mapToDomain(data);
    }
}

File: snail-race-bet/java/src/main/java/snail/race/kata/adapters/RaceResultProviderHttp.java
package snail.race.kata.adapters;

import com.fasterxml.jackson.databind.ObjectMapper;
import snail.race.kata.domain.RaceResultProvider;

import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;


public class RaceResultProviderHttp implements RaceResultProvider {
    @Override
    public SnailRaces races() {
        RaceResultProviderHttpRecords.RaceData raceData = invokeResultEndPoint();
        return RaceResultProviderHttpAnticorruptionLayer.mapToDomain(raceData);
    }

    static RaceResultProviderHttpRecords.RaceData invokeResultEndPoint() {
        try {
            var httpClient = HttpClient.newHttpClient();
            var request = HttpRequest.newBuilder(URI.create("http://localhost:8000/results/")).build();
            HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());

            return new ObjectMapper().readValue(response.body(), RaceResultProviderHttpRecords.RaceData.class);

        } catch (IOException | InterruptedException e) {
            throw new RuntimeException(e);
        }

    }

}
